internal/controllers/topology/cluster/desired_state_test.go

```diff
@@ -19,7 +19,9 @@ package cluster
 import (
 	"strings"
 	"testing"
+	"time"
 
+	"github.com/google/go-cmp/cmp"
 	. "github.com/onsi/gomega"
 	corev1 "k8s.io/api/core/v1"
```

Comment:
I don't think we're missing much with the unit test - and if that's the issue I can put the HasCP_MHC check + createMHC in a single function and test that. 

We're doing the infrastructure calculation for the CP Infrastructure at this level so I don't think we should move it down to create a single level of abstraction either - unless we take this full block and put it under computeCPDesiredState. 

Myself and @fabriziopandini went over and back on where this logic should be before - so it would be good to get his input here too.

Reply:
Ah makes sense. I missed that the controlPlaneInfraMachine is also calculated here.

Then it's probably best to keep it as is.